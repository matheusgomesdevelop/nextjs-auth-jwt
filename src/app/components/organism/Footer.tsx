import React from 'react';

import Icon from '../atoms/Icon';
import Link from '../atoms/Link';

const Footer: React.FC = () => {
    const date = new Date();

    return (
        <footer className="container flex justify-between items-center w-full">
            <p className="text-gray-500 flex gap-6">
                Â© {date.getFullYear()}
                <Link href={'https://matheusgomesdev.com.br'}>matheusgomesdev.</Link>
            </p>

            <div className="flex gap-24">
                <Link href={'https://github.com/matheusgrodrigues/nextjs-auth-jwt'}>
                    <Icon icon="pi-github" />
                </Link>

                <Link href={'https://www.linkedin.com/in/matheusgomes/'}>
                    <Icon icon="pi-linkedin" />
                </Link>
            </div>
        </footer>
    );
};

export default Footer;
