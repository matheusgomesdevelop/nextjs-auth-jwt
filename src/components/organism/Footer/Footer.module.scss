@import "../../../styles/util/_breakpoints.scss";

.footer {
  width: 100%;
  min-height: 88px;
  max-width: 1280px;
  padding: 0 var(--spacing-32);
  @include centralizar;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  gap: var(--spacing-32);

  @include d(tablet) {
    flex-direction: row;
    height: 88px;
    gap: 0;
  }

  &__name {
    & > p {
      display: flex;
      gap: var(--spacing-6);
      color: var(--gray-500);
    }
  }

  &__icons {
    display: flex;
    gap: var(--spacing-24);

    & i {
      color: var(--gray-500);
    }
  }
}
