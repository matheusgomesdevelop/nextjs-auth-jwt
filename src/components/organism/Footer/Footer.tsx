import React from 'react';

import styles from './Footer.module.scss';

import { Link, Icon } from '@/components/atoms';

interface FooterProps {
    name: string;
    site: string;
    github: string;
    linkedin: string;
}

const Footer: React.FC<FooterProps> = ({ name, site, github, linkedin }) => {
    const date = new Date();

    return (
        <div className={styles.footer} id="o-footer">
            <div className={styles.footer__name}>
                <p>
                    Â© {date.getFullYear()}
                    <Link href={site}>{name}.</Link>
                </p>
            </div>

            <div className={styles.footer__icons}>
                <Link href={github}>
                    <Icon icon="pi-github" />
                </Link>

                <Link href={linkedin}>
                    <Icon icon="pi-linkedin" />
                </Link>
            </div>
        </div>
    );
};

export default Footer;
