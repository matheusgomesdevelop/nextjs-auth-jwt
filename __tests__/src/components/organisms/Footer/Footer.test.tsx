import React from 'react';

import renderer from 'react-test-renderer';

import Footer from '@/components/organism/Footer/Footer';

describe('Deve renderizar o footer corretamente', () => {
    it('Deve preservar a estrutura visual do footer', () => {
        const three = renderer.create(<Footer />).toJSON();

        expect(three).toMatchSnapshot();
    });
});
