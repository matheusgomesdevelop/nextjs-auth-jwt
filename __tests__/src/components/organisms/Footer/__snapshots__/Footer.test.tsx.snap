// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o footer corretamente Deve preservar a estrutura visual do footer 1`] = `
<div
  className="footer"
  id="footer"
>
  <div
    className="footer__name"
  >
    <p>
      Â© 
      2024
      <a
        className="link"
        href="https://matheusgomesdev.com.br"
      >
        matheusgomesdev.
      </a>
    </p>
  </div>
  <div
    className="footer__icons"
  >
    <a
      className="link"
      href="https://github.com/matheusgrodrigues/nextjs-auth-jwt"
    >
      <i
        className="icon pi pi-github"
      />
    </a>
    <a
      className="link"
      href="https://www.linkedin.com/in/matheusgomes/"
    >
      <i
        className="icon pi pi-linkedin"
      />
    </a>
  </div>
</div>
`;
